src/core/QuoridorBoard.cs

```diff
@@ -8,26 +10,189 @@ namespace QuoridorEngine.Core
     internal class QuoridorBoard
     {
         List<(int, int)> usedWalls;
-        private int dimention;
 
-        public int Dimention { get { return dimention; } }
+        // The complete number of squares per row/column
+        // including the ones destined for walls
+        private readonly int dimension;
 
-        QuoridorBoard(int dimention)
+        public int Dimension { get { return dimension; } }
+
+        public QuoridorBoard(int dimension)
         {
             usedWalls = new List<(int, int)>();
-            this.dimention = dimention;
+            this.dimension = dimension;
+        }
+
+        /// <summary>
+        /// Utility function to check whether provided 
+        /// square coords represent valid grid square. 
+        /// </summary>
+        /// <param name="row">Square row</param>
+        /// <param name="col">Square column</param>
+        /// <returns>True if square is valid else false</returns>
+        public bool IsGridSquare(int row, int col)
+        {
+            if (row >= 2 && col >= 2 && row < dimension && col < dimension && row != col)
```

Comment:
Simply return the statement itself instead of checking with an if

return row>=2 && col >= 2.......